File: backend_db/app/models/fichaje.py
from sqlalchemy import Column, String, ForeignKey, DateTime
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from app.database import Base
from sqlalchemy import text

class Fichaje(Base):
    __tablename__ = "fichajes"

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("uuid_generate_v4()"))
    usuario_id = Column(UUID(as_uuid=True), ForeignKey("usuarios.id", ondelete="CASCADE"), nullable=False)
    empresa_id = Column(UUID(as_uuid=True), ForeignKey("empresas.id", ondelete="CASCADE"), nullable=False)
    tipo = Column(String(30), nullable=False)

    # ✅ timezone=True
    fecha_hora = Column(DateTime(timezone=True), server_default=func.now())
    fecha_creacion = Column(DateTime(timezone=True), server_default=func.now())

File: frontend/lib/models/fichaje.dart
class Fichaje {
  final String id;
  final String usuarioId;
  final String empresaId;
  final String tipo;
  final DateTime fechaHora;
  final DateTime fechaCreacion;

  Fichaje({
    required this.id,
    required this.usuarioId,
    required this.empresaId,
    required this.tipo,
    required this.fechaHora,
    required this.fechaCreacion,
  });

  factory Fichaje.fromJson(Map<String, dynamic> json) {
    return Fichaje(
      id: json['id'],
      usuarioId: json['usuario_id'],
      empresaId: json['empresa_id'],
      tipo: json['tipo'],
      fechaHora: DateTime.parse(json['fecha_hora']),
      fechaCreacion: DateTime.parse(json['fecha_creacion']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'usuario_id': usuarioId,
      'empresa_id': empresaId,
      'tipo': tipo,
      'fecha_hora': fechaHora.toIso8601String(),
      'fecha_creacion': fechaCreacion.toIso8601String(),
    };
  }
}
